sql_grader/problem.py

```diff
@@ -93,25 +111,41 @@ def clone_database(source):
         return destination
 
     @classmethod
-    def run_query(cls, source, query, verify_query=None):
+    def run_query(cls, source, query, verify_query=None,
+                  modification_query=None):
         """
-        Execute the provided SQL query against a copy of the database
+        Execute the provided answer and verification queries against a copy of
+        the database
         """
-        def run(database, query):
+        def run(database, query, is_single_query):
             result = []
             message = None
             with database as connection:
                 try:
-                    for row in connection.execute(query):
+                    if is_single_query:
+                        executor_func = connection.execute
+                    else:
+                        executor_func = connection.executescript
+                    rows = executor_func(query)
+                    for row in rows:
                         result.append(row)
                 except Exception as error:  # pylint: disable=broad-except
                     result = None
                     message = str(error)
             return result, message
+
         database = cls.clone_database(source)
-        result, error = run(database, query)
+
+        if verify_query:
```

Comment:
I know this is a textbook example of logic simplification, because it could be written as `is_single_query = not verify_query`. But the extended version makes it more clear and serves almost as an explanation. So I think it's fine.

Reply:
Yeah, I was going with `is_single_query = not verify_query` as well, but thought that this might be more explicit. :)